great_expectations/datasource/fluent/interfaces.py

```diff
@@ -462,6 +478,7 @@ def get_asset(self, asset_name: str) -> _DataAssetT:
         """Returns the DataAsset referred to by name"""
         # This default implementation will be used if protocol is inherited
         try:
+            self.assets[asset_name]._datasource = self
```

Comment:
You might have just fixed an edge case I found with my `ConfigStr` logic 😍 
It was happening because of a missing `_data_contexts.config_provider` which I didn't understand at the time.